kubectl-rook-ceph.sh

```diff
@@ -226,14 +227,220 @@ function path_cm_rook_ceph_operator_config() {
+    --id=$good_mon \
+    --foreground \
+    --public-addr=$good_mon_public_ip \
+    --setuser-match-path=/var/lib/ceph/mon/ceph-$good_mon/store.db \
+    --public-bind-addr=$ROOK_POD_IP \
+    --extract-monmap=$monmap_path
+
+  info_msg "Printing monmap"; \
+  KUBECTL_NS_CLUSTER exec deploy/rook-ceph-mon-$good_mon-debug -c mon -- monmaptool --print $monmap_path
+
+  # remove all the mons except the good one
+  for bad_mon in "${bad_mons[@]}"
+  do
+    info_msg "Removing mon $bad_mon"
+    KUBECTL_NS_CLUSTER exec deploy/rook-ceph-mon-$good_mon-debug -c mon -- monmaptool $monmap_path --rm $bad_mon
+  done
+
+  info_msg "Injecting the monmap"
+  KUBECTL_NS_CLUSTER exec deploy/rook-ceph-mon-$good_mon-debug -c mon -- ceph-mon \
+    --fsid=$ceph_fsid \
+    --keyring=/etc/ceph/keyring-store/keyring \
+    --log-to-stderr=true \
+    --err-to-stderr=true \
+    --mon-cluster-log-to-stderr=true \
+    --log-stderr-prefix=debug \
+    --default-log-to-file=false \
+    --default-mon-cluster-log-to-file=false \
+    --mon-host=$ROOK_CEPH_MON_HOST \
+    --mon-initial-members=$ROOK_CEPH_MON_INITIAL_MEMBERS \
+    --id=$good_mon \
+    --foreground \
+    --public-addr=$good_mon_public_ip \
+    --setuser-match-path=/var/lib/ceph/mon/ceph-$good_mon/store.db \
+    --public-bind-addr=$ROOK_POD_IP \
+    --inject-monmap=$monmap_path
+  info_msg "Finished updating the monmap!"
+  set -eu
+
+  info_msg "Printing final monmap"
+  KUBECTL_NS_CLUSTER exec deploy/rook-ceph-mon-$good_mon-debug -c mon -- monmaptool --print $monmap_path
+
+  info_msg "Restoring the mons in the rook-ceph-mon-endpoints configmap to the good mon"
+  KUBECTL_NS_CLUSTER patch configmaps rook-ceph-mon-endpoints --type json --patch "[{ op: replace, path: /data/data, value: $good_mon=$good_mon_public_ip:$good_mon_port }]"
+
+  info_msg "Stopping the debug pod for mon $good_mon"
+  run_stop_debug rook-ceph-mon-$good_mon
+
+  info_msg "Check that the restored mon is responding"
+  wait_for_mon_status_response
+
+  info_msg "Purging the bad mons: ${bad_mons[*]}"
+  for bad_mon in "${bad_mons[@]}"
+  do
+    info_msg "purging old mon: $bad_mon"
+    KUBECTL_NS_CLUSTER delete deploy rook-ceph-mon-$bad_mon
+    KUBECTL_NS_CLUSTER delete svc rook-ceph-mon-$bad_mon
+  done
+
+  info_msg "Mon quorum was successfully restored to mon $good_mon"
+
+  prompt_to_continue_or_cancel "Start up the operator and expand to full mon quorum again?" "yes"
+
+  # scale up the operator
+  KUBECTL_NS_OPERATOR scale deployment rook-ceph-operator --replicas=1
+}
+
+function prompt_to_continue_or_cancel() {
+  proceed_question=$1
+  proceed_answer=$2
+  info_msg "$proceed_question If so, enter: $proceed_answer"
+  set +u
+  if [[ "$ROOK_PLUGIN_SKIP_PROMPTS" != "true" ]]; then
+    read INPUT_VAR
+    if [ "$INPUT_VAR" = "$proceed_answer" ]; then
+      info_msg "proceeding"
+    else
+      warn_msg "cancelled"
+      exit 1
+    fi
+  else
+    warn_msg "skipped prompt since ROOK_PLUGIN_SKIP_PROMPTS=true"
+  fi
```

Comment:
```suggestion
  else
    warn_msg "skipped prompt since ROOK_PLUGIN_SKIP_PROMPTS=true"
  fi
```

is this set to handle for CI as I think it is good to have user input everytime?

Reply:
Yes we need it for CI, and could be useful if someone automates it